### PR TITLE
Re-add legacy builds and lock it behind a quiz

### PR DESCRIPTION
**Pull request by clrxbl**

### CODE REVIEW

**Comment by electronicboy** [comment]
No from me unless there is a mechanism where people have to explicitly acknowledge that these builds are unsupported in every form, shape, or manner; the string I talked about was only 50% of a gag. people seem to just skip past all of the reading aspect, so, having an action they have to explicitly take, especially if it's one where they have to type it, at least for me, is a massive win

**Comment by kennytv** [comment]
I'd like for that to be a lite version of, so people **really** know: https://protocol.support/old/ ψ(._. )>

**Review by kennytv — DISMISSED** [review]
(Properly blocking, really want to see that *actually* happen in a similar way where you have to type or select one or two options/sentences acknowledging you won't get support. But that'll have to be more than just a checkbox)

**Comment by NoahvdAa** [comment]
I prefer this solution:



In all seriousness, something like kenny linked (but slightly less extreme) is probably a good solution. It'll allow users that *really, really* want to download legacy builds to download them, and at the same time hopefully keep people from using legacy versions more. Maybe have them type a sentence? (looks like kenny mentioned this right before I clicked comment)

**Comment by Phoenix616** [comment]
I feel like simply putting in a check box besides a "I acknowledge that by downloading this build I will receive no support and that running it might expose me to unpatched bugs and exploits." sentence should be enough.

Going through extreme lengths to download something will at best just get people to ignore what is said there or at worst make them go to a third party site which distributed jars with questionable content. At least with the checkbox there is a clear user action required (which at least seems to hold up legally). Will it stop people from using those versions? No. But so doesn't any other method which still lets one download it.

**Comment by electronicboy** [comment]
We've had the message saying that we offer 0 support for these versions and we still get the expectation of support. We don't care if people use them, we're just tired of the expectation that they're still supported because we offer them

**Comment by kennytv** [comment]
A checkbox would work just as well as any checklist people tick in GitHub issues saying "I'm on latest, I've followed the issue template", especially with the kind of people asking for legacy support

**Comment by Phoenix616** [comment]
> A checkbox would work just as well as any checklist people tick in GitHub issues saying "I'm on latest, I've followed the issue template", especially with the kind of people asking for legacy support. It's not like the legacy page is the main attraction Paper has to offer that thousands of people use on a daily basis

It will work as well as any other method which is basically that it doesn't. The only way to completely distance Paper from the old versions is to not offer those jars in the first place on the website itself and live with the fact that people are going to use random sources.

**Comment by clrxbl** [comment]
I have added my magnum opus in web development:

https://user-images.githubusercontent.com/28601081/145722390-5d65f89c-0690-4b4b-b5c0-9de9874fe451.mp4

**Comment by electronicboy** [comment]
image

File: src/js/legacy.js [hunk 1]
```diff
@@ -76,6 +74,67 @@ const downloads = {
...
+  } else {
+    return false;
+  }
+}
+
+submitButton.onclick = function() {
+  submitButton.disabled = true;
+  let failed = false;
+
+  if (! checkValue("quiz-1", "2")) {
```
Comment (NoahvdAa) [inline src/js/legacy.js:101]: Can't we put all of these into one if statement?

Comment (JRoy) [reply src/js/legacy.js:101]: it's impossible, could never be done

Comment (NoahvdAa) [reply src/js/legacy.js:101]: Thanks for checking that for me, sorry for making impossible requests :(

Comment (clrxbl) [reply src/js/legacy.js:101]: I've compacted the quiz answer stuff into one else if statement but other than that, don't think so?

Comment (NoahvdAa) [reply src/js/legacy.js:101]: You can compact it into a single if statement using the lovely OR operator:
```js
if (!checkValue("quiz-1", "2") || !checkValue("quiz-2", "2") || !checkValue("quiz-3", "2") || !checkValue("quiz-4", "4")) {
       failed = true;
}
```

Comment (clrxbl) [reply src/js/legacy.js:101]: Doesn't that become ugly if there's a lot more than 4 questions? (Unsure if the questions we have right now are the ones we'll want to be asking at the moment)

Comment (NoahvdAa) [reply src/js/legacy.js:101]: IMO this is a lot cleaner than having multiple lines of nearly identical if statements

Comment (clrxbl) [reply src/js/legacy.js]: I've gone with your route for now at least, thanks.

**Review by kennytv — APPROVED** [review]

**Comment by mbax** [comment]
Why are the questions *so incredibly tiny*?

**Comment by clrxbl** [comment]
> Why are the questions _so incredibly tiny_?

Already resolved
image

**Review by NotMyFault — CHANGES_REQUESTED** [review]
The font is barely visible if you're using different themes.

**Comment by clrxbl** [comment]
>  The font is barely visible if you're using different themes.

I forgot that the Paper site can be viewed in light mode since there's no light/dark mode toggle on the site. Fixed.

**Comment by NotMyFault** [comment]
> >  The font is barely visible if you're using different themes.
> 
> I forgot that the Paper site can be viewed in light mode since there's no light/dark mode toggle on the site. Fixed.

Looking better now, thanks.

**Comment by Cryptite** [comment]
> we're just tired of the expectation that they're still supported because we offer them

To be fair this is an eternal, and permanent constant. This will never change. As long as you have the CYA bit anywhere, you can always point to it.

**Review by JRoy — APPROVED** [review]
works as intended and uses non-free software

File: src/legacy.twig [hunk 2]
```diff
@@ -9,7 +9,81 @@
...
+          <i class="tiny material-icons">error</i>
+          Any problems you find in these builds <strong>will not be fixed</strong>.
+      </li>
+      <li>
+          <i class="tiny material-icons">error</i>
+          Any issues you report for these builds <strong>will be closed</strong>.
+      </li>
+      <li>
+        <i class="tiny material-icons">error</i>
+          You may be <strong>kicked or muted</strong> when asking for support on our Discord.
```
Comment (TwoLeggedCat) [inline src/legacy.twig:28]: Bit late to the party, but perhaps this (and the form) should mention the forums as well?

**Comment by MWHunter** [comment]
Failing the quiz multiple times makes the countdown faster.

**Pull request merged**